### DISCUSSION TITLE
Retrieve download information for a crate within a specified timeframe.

### DISCUSSION THREAD

**Discussion by dxphilo** [rust-lang/crates.io#7973] [Q&A] [answered]
I've been exploring the crates.io API and was curious if there's an endpoint available to retrieve crate download data within a specified timeframe, such as a start and end date.

Been looking over at [https://crates.io/data-access](https://crates.io/data-access) but can't find enough information. Thanks.

**Comment by LawnGnome** [comment, marked as answer]
The last 90 days of per-day, per-crate-version download counts are included in the nightly database dumps (linked from the data access page) in the `version_downloads` table.

For a small number of crates, you could also use the download API endpoint, eg https://crates.io/api/v1/crates/syn/downloads — while this isn't technically documented, it should provide the same information on a per-crate basis, and is unlikely to change in practice any time soon.

If you need more historical data, let us know what you need at help@crates.io and we can see if we can provide it. We do have extra history at the same granularity (per-day, per-crate-version) available; we just don't generally provide it because, honestly, it's not that useful to most people and it's a lot of raw data to serve up.

**Reply by Turbo87** [reply]
there is now also https://static.crates.io/archive/version-downloads/ which hosts all of the download numbers dating back to the beginning of crates.io

**Reply by dxphilo** [reply]
Ooooh God. Thanks a bunch. Got some project that was stalled by this seems I am gonnah brush it up. Ill share it up.

Thanks for getting by.
